### DISCUSSION TITLE
Please make it easier to insert new sprints in the middle

### DISCUSSION THREAD

**Discussion by AlekSi** [community/community#42491] [Projects and Issues]
### Select Topic Area

Product Feedback

### Body

We use new Projects with a Sprint field to track our sprints and backlog. Our main dashboard looks like that:

<img width="450" alt="CleanShot 2022-12-23 at 19 07 47@2x" src="https://user-images.githubusercontent.com/11512/209357339-d13f382e-3a0e-45b5-8add-bb018c65d9d7.png">

Here we have the current sprint, the next planned sprint, the main backlog, the backlog of estimated tasks, the preparation backlog, and the rest. Both "Core backlog" and "Core preparation" are Sprint field values, and their dates a fake.

Now, I want to create Sprint 29. And can't do that easily:

<img width="427" alt="CleanShot 2022-12-23 at 19 08 46@2x" src="https://user-images.githubusercontent.com/11512/209357452-5e0b86ef-14f2-473f-b805-1adab770349b.png">

Sprint 29 is added at the end, and I can't change its start date to anything earlier than January 16. If I edit the Core backlog's dates…

<img width="667" alt="CleanShot 2022-12-23 at 19 11 55@2x" src="https://user-images.githubusercontent.com/11512/209357937-c3f86db4-4c19-44ce-804c-f324eab2f22d.png">

Everything shifts by one week. I can't replace the break with a sprint. If I delete this new break, everything returns to how it was before.

The only option for me now is to rename "Core backlog" and "Core preparation" to "Sprint 29" and "Sprint 30", add new "Core backlog" and "Core preparation" "sprints", and then move issues manually. Given that those two "sprint" are prioritized by simple ordering – that's a lot of manual and error-prone labor.

I think a part of the problem is that validation happens when something is added, removed, or edited, not when the "Save changes" button is clicked. That ensures that all dates are "valid", but doesn't allow changes I want to make. I think removing that automation and replacing it with warnings, and disabling saving until everything is "valid" would resolve my problem.

Another option is just™ to allow inserting sprints just like it is possible to insert breaks right now.

**Comment by AlekSi** [comment]
Ack?

**Comment by taylorkaren** [comment]
Having the same problem and this is really frustrating. What Alekso has outlined above is a severe hindrance to managing buckets of work that are not yet assigned to a sprint, as well as more or less requiring you to enter sprints into perpetuity so that you don't have to mess with this.

**Comment by AlekSi** [comment]
Is anyone from GitHub reading those?

**Comment by tafelnl** [comment]
> I think a part of the problem is that validation happens when something is added, removed, or edited, not when the "Save changes" button is clicked. That ensures that all dates are "valid", but doesn't allow changes I want to make. I think removing that automation and replacing it with warnings, and disabling saving until everything is "valid" would resolve my problem.

> Another option is just™ to allow inserting sprints just like it is possible to insert breaks right now.

^ this is spot on. Implementing both of these changes would make a lot of sense UX wise. You shouldn't restrict a user on what he can do until actually hitting the save button, because you would then end up with weird 'edge' cases like the one we're experiencing now

**Comment by kennethjohnson2452-cloud** [comment]
That makes sense, the current sprint validation feels too rigid for workflows that need occasional restructuring. A similar flexibility matters in areas like [law firm search engine optimisation](https://pearllemonseo.co.uk/industries/law-firms/), where changing priorities shouldn’t force unnecessary manual work.
